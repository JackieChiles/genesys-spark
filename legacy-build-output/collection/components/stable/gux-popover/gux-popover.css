:host {
  color: #2e394c;
}
.gux-popover-wrapper {
  z-index: var(--gux-zindex-popover, 2);
  display: inline-block;
  padding: 16px;
  background-color: #fdfdfd;
  border-radius: 4px;
  box-shadow: 0 3px 14px rgba(46, 57, 76, 0.22);
}
.gux-popover-wrapper .gux-arrow,
.gux-popover-wrapper .gux-arrow::before {
  position: absolute;
  width: 10px;
  height: 10px;
  background: inherit;
}
.gux-popover-wrapper .gux-arrow {
  visibility: hidden;
}
.gux-popover-wrapper .gux-arrow::before {
  visibility: visible;
  content: '';
  transform: rotate(45deg);
}
.gux-popover-wrapper[data-popper-placement='bottom'] .gux-arrow,
.gux-popover-wrapper[data-popper-placement='bottom-start'] .gux-arrow,
.gux-popover-wrapper[data-popper-placement='bottom-end'] .gux-arrow {
  top: -5px;
}
.gux-popover-wrapper[data-popper-placement='top'] .gux-arrow,
.gux-popover-wrapper[data-popper-placement='top-start'] .gux-arrow,
.gux-popover-wrapper[data-popper-placement='top-end'] .gux-arrow {
  bottom: -5px;
}
.gux-popover-wrapper[data-popper-placement='left'] .gux-arrow,
.gux-popover-wrapper[data-popper-placement='left-start'] .gux-arrow,
.gux-popover-wrapper[data-popper-placement='left-end'] .gux-arrow {
  right: -5px;
}
.gux-popover-wrapper[data-popper-placement='right'] .gux-arrow,
.gux-popover-wrapper[data-popper-placement='right-start'] .gux-arrow,
.gux-popover-wrapper[data-popper-placement='right-end'] .gux-arrow {
  left: -5px;
}
.gux-popover-wrapper .gux-popover-header {
  font-size: 12px;
  line-height: 20px;
  font-family: Roboto, sans-serif;
  font-weight: 400;
  font-weight: 700;
  display: flex;
  flex-direction: row;
  align-content: center;
  align-items: center;
  justify-content: space-between;
  padding-bottom: 16px;
}
.gux-popover-wrapper gux-dismiss-button {
  float: right;
}
