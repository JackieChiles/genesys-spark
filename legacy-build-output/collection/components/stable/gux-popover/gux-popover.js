var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
  var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
  if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
    r = Reflect.decorate(decorators, target, key, desc);
  else
    for (var i = decorators.length - 1; i >= 0; i--)
      if (d = decorators[i])
        r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
  return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { createPopper } from '@popperjs/core';
import { h } from '@stencil/core';
import { OnClickOutside } from '../../../utils/decorator/on-click-outside';
import { onHiddenChange } from '../../../utils/dom/on-attribute-change';
import { trackComponent } from '@utils/tracking/usage';
import { getSlot } from '@utils/dom/get-slot';
/**
 * @slot - popover content
 * @slot title - Slot for popover title
 */
export class GuxPopover {
  constructor() {
    this.for = undefined;
    this.position = 'bottom';
    this.displayDismissButton = undefined;
    this.closeOnClickOutside = false;
    this.hidden = true;
  }
  hiddenHandler(hidden) {
    if (!hidden && !this.popperInstance) {
      this.runPopper();
    }
    else if (!hidden && this.popperInstance) {
      this.popperInstance.forceUpdate();
    }
  }
  checkForClickOutside(event) {
    const clickPath = event.composedPath();
    const forElement = document.getElementById(this.for);
    const clickedForElement = clickPath.includes(forElement);
    if ((this.closeOnClickOutside || !this.displayDismissButton) &&
      !this.hidden &&
      !clickedForElement) {
      this.dismiss();
    }
  }
  get titleSlot() {
    return getSlot(this.root, 'title');
  }
  runPopper() {
    const forElement = document.getElementById(this.for);
    if (!forElement) {
      console.error(`gux-popover: invalid "for" attribute. No element in page with the id "${this.for}"`);
    }
    else if (this.popupElement) {
      this.popperInstance = createPopper(forElement, this.popupElement, {
        modifiers: [
          {
            name: 'computeStyles',
            options: {
              gpuAcceleration: false
            }
          },
          {
            name: 'offset',
            options: {
              offset: [0, 7]
            }
          },
          {
            name: 'arrow',
            options: {
              padding: 16
            }
          }
        ],
        placement: this.position
      });
    }
  }
  destroyPopper() {
    if (this.popperInstance) {
      this.popperInstance.destroy();
      this.popperInstance = null;
    }
  }
  dismiss() {
    const dismissEvent = this.guxdismiss.emit();
    if (!dismissEvent.defaultPrevented) {
      this.root.setAttribute('hidden', '');
    }
  }
  connectedCallback() {
    trackComponent(this.root, { variant: this.position });
    this.hiddenObserver = onHiddenChange(this.root, (hidden) => {
      this.hidden = hidden;
    });
    this.hidden = this.root.hidden;
  }
  componentDidLoad() {
    this.runPopper();
  }
  disconnectedCallback() {
    this.destroyPopper();
    if (this.hiddenObserver) {
      this.hiddenObserver.disconnect();
    }
  }
  renderDismissButton() {
    if (this.displayDismissButton) {
      return (h("gux-dismiss-button", { onClick: this.dismiss.bind(this), position: "inherit" }));
    }
  }
  render() {
    return (h("div", { ref: (el) => (this.popupElement = el), class: "gux-popover-wrapper" }, h("div", { class: "gux-arrow", "data-popper-arrow": true }), h("div", { class: { 'gux-popover-header': Boolean(this.titleSlot) } }, h("slot", { name: "title" }), this.renderDismissButton()), h("div", { class: "gux-popover-content" }, h("slot", null))));
  }
  static get is() { return "gux-popover"; }
  static get encapsulation() { return "shadow"; }
  static get originalStyleUrls() {
    return {
      "$": ["gux-popover.less"]
    };
  }
  static get styleUrls() {
    return {
      "$": ["gux-popover.css"]
    };
  }
  static get properties() {
    return {
      "for": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "string",
          "resolved": "string",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": "Indicates the id of the element the popover should anchor to"
        },
        "attribute": "for",
        "reflect": false
      },
      "position": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "PopperPosition",
          "resolved": "\"bottom\" | \"bottom-end\" | \"bottom-start\" | \"left\" | \"left-end\" | \"left-start\" | \"right\" | \"right-end\" | \"right-start\" | \"top\" | \"top-end\" | \"top-start\"",
          "references": {
            "PopperPosition": {
              "location": "import",
              "path": "./gux-popover.types"
            }
          }
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": "Indicate position of popover element arrow (follow popper js position attribute api)"
        },
        "attribute": "position",
        "reflect": false,
        "defaultValue": "'bottom'"
      },
      "displayDismissButton": {
        "type": "boolean",
        "mutable": false,
        "complexType": {
          "original": "boolean",
          "resolved": "boolean",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": "Indicate if the dismiss button is displayed"
        },
        "attribute": "display-dismiss-button",
        "reflect": false
      },
      "closeOnClickOutside": {
        "type": "boolean",
        "mutable": false,
        "complexType": {
          "original": "boolean",
          "resolved": "boolean",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": "Close popover when the user clicks outside of its bounds"
        },
        "attribute": "close-on-click-outside",
        "reflect": false,
        "defaultValue": "false"
      }
    };
  }
  static get states() {
    return {
      "hidden": {}
    };
  }
  static get events() {
    return [{
        "method": "guxdismiss",
        "name": "guxdismiss",
        "bubbles": true,
        "cancelable": true,
        "composed": true,
        "docs": {
          "tags": [],
          "text": "Fired when a user dismisses the popover"
        },
        "complexType": {
          "original": "void",
          "resolved": "void",
          "references": {}
        }
      }];
  }
  static get elementRef() { return "root"; }
  static get watchers() {
    return [{
        "propName": "hidden",
        "methodName": "hiddenHandler"
      }];
  }
}
__decorate([
  OnClickOutside({ triggerEvents: 'mousedown' })
], GuxPopover.prototype, "checkForClickOutside", null);
