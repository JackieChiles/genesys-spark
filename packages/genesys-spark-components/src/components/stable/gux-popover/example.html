<h1>gux-popover</h1>
<div class="scroll-container">
  <div class="scroll-interior">
    <gux-button id="popover-target">Example Element</gux-button>
    <gux-popover
      id="popover-example"
      position="top"
      for="popover-target"
      is-open="false"
      displayDismissButton="true"
    >
      <span slot="title">Title</span>
      <div class="popover-content-wrapper">
        <div>Popover Content</div>
        <gux-cta-group align="end">
          <gux-button slot="primary" onclick="notify(event)"
            >Action 3</gux-button
          >
          <gux-button slot="secondary" onclick="notify(event)"
            >Action 2</gux-button
          >
          <gux-button slot="dismiss" onclick="notify(event)"
            >Action 1</gux-button
          >
        </gux-cta-group>
      </div>
    </gux-popover>
  </div>
</div>

<style>
  :popover-open {
    position: absolute;
    inset: unset;
    top: 5px;
    width: 200px;
    height: 100px;
    margin: 0;
  }

  .scroll-container {
    width: 900px;
    height: 300px;
    margin: auto;
    overflow: scroll;
    border: 1px solid #aaa;
  }

  .scroll-interior {
    position: relative;
    width: 600px;
    height: 600px;
    background-color: #fff;
  }

  .popover-content-wrapper {
    display: flex;
    flex-direction: column;
    gap: 16px;
  }

  .popover-content {
    display: flex;
    flex-direction: row;
    gap: 8px;
    width: 280px;
  }

  #popover-target {
    position: absolute;
    left: 250px;
    display: inline-block;
    padding: 10px 15px;
    background-color: #fdfdfd;
    border: 1px #e4e9f0;
    box-shadow: 0 0 2px fade(#202937, 24%);
  }
</style>
