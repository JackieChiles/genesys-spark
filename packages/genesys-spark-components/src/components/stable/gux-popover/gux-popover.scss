@import '../../../style/zindex.scss';

.gux-popover-wrapper {
  position: absolute;
  top: 0;
  left: 0;
  min-width: 280px;
  z-index: var(--gux-zindex-popover, $gux-zindex-popover);
  display: inline-block;
  padding: var(--gse-ui-popover-gap);
  background-color: var(--gse-ui-popover-backgroundColor);
  border-radius: var(--gse-ui-popover-borderRadius);
  box-shadow: var(--gse-ui-popover-boxShadow-x)
    var(--gse-ui-popover-boxShadow-y) var(--gse-ui-popover-boxShadow-blur)
    var(--gse-ui-popover-boxShadow-spread) var(--gse-ui-tooltip-boxShadow-color);

  &.gux-hidden {
    display: none;
  }

  .gux-arrow {
    position: absolute;
    width: var(--gse-ui-popover-anchor-width);
    height: var(--gse-ui-popover-anchor-width);
    background: var(--gse-ui-popover-backgroundColor);
  }

  .gux-popover-header {
    font-family: var(--gse-ui-popover-title-text-fontFamily);
    font-size: var(--gse-ui-popover-title-text-fontSize);
    font-weight: var(--gse-ui-popover-title-text-fontWeight);
    line-height: var(--gse-ui-popover-title-text-lineHeight);
    color: (--gse-ui-popover-headerColor);
    display: flex;
    flex-direction: row;
    align-content: center;
    align-items: center;
    justify-content: space-between;
    padding-bottom: var(--gse-ui-popover-gap);
  }

  gux-dismiss-button {
    float: right;
  }
}
