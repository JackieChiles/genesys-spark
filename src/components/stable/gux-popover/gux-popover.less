@import (reference) '../../../style/color.less';
@import (reference) '../../../style/mixins.less';

// Style
gux-popover {
  z-index: 2;
  color: @gux-type;

  .gux-popover-wrapper {
    display: flex;
    flex-direction: column;
    background-color: @gux-type-dark;
    border: 1px solid @gux-grey-30;
    border-radius: 4px;
    .shadow-20();
  }

  .gux-arrow,
  .gux-arrow::after {
    position: absolute;
    width: 0;
    height: 0;
    content: ' ';
    border-color: transparent;
    border-style: solid;
  }

  .gux-arrow {
    border-width: 6px;
  }

  .gux-arrow::after {
    border-width: 4px;
  }

  &[data-popper-placement='bottom'],
  &[data-popper-placement='bottom-start'],
  &[data-popper-placement='bottom-end'] {
    margin-top: 6px;

    .gux-arrow {
      margin-top: -12px;
      border-bottom-color: @gux-grey-30;

      &::after {
        margin-top: -2px;
        margin-left: -4px;
        border-bottom-color: @gux-type-dark;
      }
    }
  }

  &[data-popper-placement='top'],
  &[data-popper-placement='top-start'],
  &[data-popper-placement='top-end'] {
    margin-bottom: 6px;

    .gux-arrow {
      bottom: 0;
      margin-bottom: -11px;
      border-top-color: @gux-grey-30;

      &::after {
        margin-top: -7px;
        margin-left: -4px;
        border-top-color: @gux-type-dark;
      }
    }
  }

  &[data-popper-placement='left'],
  &[data-popper-placement='left-start'],
  &[data-popper-placement='left-end'] {
    margin-right: 6px;

    .gux-arrow {
      right: 0;
      margin-right: -11px;
      border-left-color: @gux-grey-30;

      &::after {
        margin-top: -4px;
        margin-left: -7px;
        border-left-color: @gux-type-dark;
      }
    }
  }

  &[data-popper-placement='right'],
  &[data-popper-placement='right-start'],
  &[data-popper-placement='right-end'] {
    margin-left: 6px;

    .gux-arrow {
      margin-left: -12px;
      border-right-color: @gux-grey-30;

      &::after {
        margin-top: -4px;
        margin-left: -2px;
        border-right-color: @gux-type-dark;
      }
    }
  }

  .gux-title-bar {
    align-self: flex-end;
    margin-bottom: -16px;

    .gux-dismiss {
      width: 12px;
      height: 12px;
      padding: 4px 4px 0 0;
      color: @gux-type;
      cursor: pointer;

      &:hover {
        color: @gux-blue-60;
      }
    }
  }

  .gux-popover-content {
    margin: 16px 18px;
  }
}
