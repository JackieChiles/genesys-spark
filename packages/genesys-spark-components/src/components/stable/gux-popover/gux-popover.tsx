import { createPopper, Instance } from '@popperjs/core';
import {
  Component,
  Element,
  Event,
  EventEmitter,
  h,
  JSX,
  Prop,
  State,
  Watch
} from '@stencil/core';

import { OnClickOutside } from '../../../utils/decorator/on-click-outside';
import { onHiddenChange } from '../../../utils/dom/on-attribute-change';
import { trackComponent } from '@utils/tracking/usage';

import { PopperPosition } from './gux-popover.types';
import { getSlot } from '@utils/dom/get-slot';
import { findElementById } from '@utils/dom/find-element-by-id';

/**
 * @slot - popover content
 * @slot title - Slot for popover title
 */

@Component({
  styleUrl: 'gux-popover.less',
  tag: 'gux-popover',
  shadow: true
})
export class GuxPopover {
  private popperInstance: Instance;
  private hiddenObserver: MutationObserver;
  private popupElement: HTMLDivElement;

  @Element()
  private root: HTMLElement;

  /**
   * Indicates the id of the element the popover should anchor to
   */
  @Prop()
  for: string;

  /**
   * Indicate position of popover element arrow (follow popper js position attribute api)
   */
  @Prop()
  position: PopperPosition = 'bottom';

  /**
   * Indicate if the dismiss button is displayed
   */
  @Prop()
  displayDismissButton: boolean;

  /**
   * Close popover when the user clicks outside of its bounds
   */
  @Prop()
  closeOnClickOutside: boolean = false;

  /**
   * Fired when a user dismisses the popover
   */
  @Event()
  guxdismiss: EventEmitter<void>;

  @State()
  hidden: boolean = true;

  @Watch('hidden')
  hiddenHandler(hidden: boolean) {
    if (!hidden && !this.popperInstance) {
      this.runPopper();
    } else if (!hidden && this.popperInstance) {
      this.popperInstance.forceUpdate();
    }
  }

  @OnClickOutside({ triggerEvents: 'mousedown' })
  checkForClickOutside(event: MouseEvent) {
    const clickPath = event.composedPath();
    const forElement = findElementById(this.root, this.for);
    const clickedForElement = clickPath.includes(forElement);

    if (
      (this.closeOnClickOutside || !this.displayDismissButton) &&
      !this.hidden &&
      !clickedForElement
    ) {
      this.dismiss();
    }
  }

  get titleSlot(): HTMLSlotElement | null {
    return getSlot(this.root, 'title');
  }

  private runPopper(): void {
    const forElement = findElementById(this.root, this.for);

    if (!forElement) {
      console.error(
        `gux-popover: invalid "for" attribute. No element in page with the id "${this.for}"`
      );
    } else if (this.popupElement) {
      this.popperInstance = createPopper(forElement, this.popupElement, {
        modifiers: [
          {
            name: 'computeStyles',
            options: {
              gpuAcceleration: false
            }
          },
          {
            name: 'offset',
            options: {
              offset: [0, 7]
            }
          },
          {
            name: 'arrow',
            options: {
              padding: 16
            }
          }
        ],
        placement: this.position
      });
    }
  }

  private destroyPopper(): void {
    if (this.popperInstance) {
      this.popperInstance.destroy();
      this.popperInstance = null;
    }
  }

  private dismiss(): void {
    const dismissEvent = this.guxdismiss.emit();
    if (!dismissEvent.defaultPrevented) {
      this.root.setAttribute('hidden', '');
    }
  }

  connectedCallback(): void {
    trackComponent(this.root, { variant: this.position });

    this.hiddenObserver = onHiddenChange(this.root, (hidden: boolean) => {
      this.hidden = hidden;
    });

    this.hidden = this.root.hidden;
  }

  componentDidLoad(): void {
    this.runPopper();
  }

  disconnectedCallback(): void {
    this.destroyPopper();
    if (this.hiddenObserver) {
      this.hiddenObserver.disconnect();
    }
  }

  private renderDismissButton(): JSX.Element {
    if (this.displayDismissButton) {
      return (
        <gux-dismiss-button
          onClick={this.dismiss.bind(this)}
          position="inherit"
        ></gux-dismiss-button>
      ) as JSX.Element;
    }
  }

  render(): JSX.Element {
    return (
      <div
        ref={(el: HTMLDivElement) => (this.popupElement = el)}
        class="gux-popover-wrapper"
      >
        <div class="gux-arrow" data-popper-arrow />
        <div class={{ 'gux-popover-header': Boolean(this.titleSlot) }}>
          <slot name="title"></slot>
          {this.renderDismissButton()}
        </div>
        <div class="gux-popover-content">
          <slot />
        </div>
      </div>
    ) as JSX.Element;
  }
}
