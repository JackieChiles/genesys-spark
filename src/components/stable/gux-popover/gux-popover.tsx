import { createPopper, Instance } from '@popperjs/core';
import {
  Component,
  Element,
  Event,
  EventEmitter,
  h,
  JSX,
  Listen,
  Prop,
  State,
  Watch
} from '@stencil/core';
import { ClickOutside } from 'stencil-click-outside';

import { buildI18nForComponent, GetI18nValue } from '../../../i18n';
import onHiddenChange from '../../../common-utils';

import modalComponentResources from './i18n/en.json';
import { PopperPosition } from './gux-popover.types';

@Component({
  styleUrl: 'gux-popover.less',
  tag: 'gux-popover'
})
export class GuxPopover {
  private i18n: GetI18nValue;
  private popperInstance: Instance;
  private forElement: HTMLElement;

  @Element() private element: HTMLElement;

  /**
   * Indicates the id of the element the popover should anchor to
   */
  @Prop()
  for: string;

  /**
   * Indicate position of popover element arrow (follow popper js position attribute api)
   */
  @Prop()
  position: PopperPosition = 'bottom';

  /**
   * Indicate if the dismiss button is displayed
   */
  @Prop()
  displayDismissButton: boolean;

  /**
   * Close popover when the user clicks outside of its bounds
   */
  @Prop()
  closeOnClickOutside: boolean = false;

  /**
   * Fired when a user dismisses the popover
   */
  @Event()
  guxdismiss: EventEmitter<void>;

  @Watch('hidden')
  watchHidden() {
    if (this.popperInstance) {
      this.popperInstance.forceUpdate();
    }
  }

  @State()
  hidden: boolean = true;

  @Listen('click', { target: 'window' })
  onClickAway(e: FocusEvent) {
    if (!this.displayDismissButton && !this.hidden) {
      if (!e.relatedTarget || !this.element.contains(e.relatedTarget as Node)) {
        this.dismiss();
      }
    }
  }

  @Watch('hidden')
  hiddenHandler(hidden: boolean) {
    if (!hidden && !this.popperInstance) {
      this.runPopper();
    }
  }

  @ClickOutside()
  checkForClickOutside() {
    if (this.closeOnClickOutside && !this.hidden) {
      this.dismiss();
    }
  }

  private runPopper(): void {
    if (this.forElement) {
      this.popperInstance = createPopper(this.forElement, this.element, {
        modifiers: [
          {
            name: 'offset',
            options: {
              offset: [0, 7]
            }
          }
        ],
        placement: this.position
      });
    } else {
      console.error(
        `GUX-Popover: invalid element supplied to 'for': "${this.for}"`
      );
    }
  }

  private destroyPopper(): void {
    if (this.popperInstance) {
      this.popperInstance.destroy();
      this.popperInstance = null;
    }
  }

  private dismiss(): void {
    const dismissEvent = this.guxdismiss.emit();
    if (!dismissEvent.defaultPrevented) {
      this.element.setAttribute('hidden', '');
    }
  }

  async componentWillLoad(): Promise<void> {
    this.forElement = document.getElementById(this.for);

    this.i18n = await buildI18nForComponent(
      this.element,
      modalComponentResources
    );

    onHiddenChange(this.element, (hidden: boolean) => {
      this.hidden = hidden;
    });

    this.hidden = this.element.hidden;
  }

  componentDidLoad(): void {
    this.runPopper();
  }

  componentDidUnload(): void {
    this.destroyPopper();
  }

  render(): JSX.Element {
    return (
      <div class={`popover-wrapper`}>
        <div class="arrow" data-popper-arrow />
        {this.displayDismissButton && (
          <div class="title-bar">
            <gux-icon
              class="dismiss"
              icon-name="close"
              screenreader-text={this.i18n('dismiss')}
              onClick={this.dismiss.bind(this)}
            />
          </div>
        )}
        <div class="popover-content">
          <slot />
        </div>
      </div>
    );
  }
}
