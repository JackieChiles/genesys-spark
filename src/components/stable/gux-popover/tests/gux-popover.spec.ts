import { newSpecPage } from '@stencil/core/testing';
import * as popperjs from '@popperjs/core';
import MutationObserver from 'mutation-observer';
import { GuxPopover } from '../gux-popover';

const components = [GuxPopover];
const language = 'en';

describe('gux-popover', () => {
  beforeEach(async () => {
    global.MutationObserver = MutationObserver;
    // popperjs does not work with Stencils MockHTMLElements used in tests
    jest.spyOn(popperjs, 'createPopper').mockReturnValue({
      destroy: jest.fn()
    } as unknown as popperjs.Instance);
  });

  afterEach(async () => {
    jest.spyOn(popperjs, 'createPopper').mockRestore();
  });

  describe('#render', () => {
    [
      {
        description: 'should render popover',
        html: `
        <div>
        <div id="popover-target">
          Example Element
        </div>
        <gux-popover id="popover-example" position="top" for="popover-target">
          <span slot="title">Title</span>
          <div>popover content</div>
        </gux-popover>
      </div>
        `
      }
    ].forEach(({ description, html }) => {
      it(description, async () => {
        const page = await newSpecPage({ components, html, language });

        expect(page.rootInstance).toBeInstanceOf(GuxPopover);

        expect(page.root).toMatchSnapshot();
      });
    });
  });
});
