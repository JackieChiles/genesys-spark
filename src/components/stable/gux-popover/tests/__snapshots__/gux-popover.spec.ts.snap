// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-popover #render should render popover 1`] = `
<gux-popover for="popover-target" id="popover-example" position="top">
  <!---->
  <div class="popover-wrapper">
    <div class="arrow" data-popper-arrow></div>
    <div class="popover-content">
      <div>
        popover content
      </div>
    </div>
  </div>
</gux-popover>
`;
